### PR TITLE
Fix build with FFmpeg master. Some deprecated APIs have been removed.

### PR DESCRIPTION
**Pull request by jviney**
OpenCV currently does not build with FFmpeg master. This is because 2 of their deprecated APIs have been removed.

- `AVFMT_RAWPICTURE` removed
- Non `AV_` prefixed avcodec flags have been removed

### CODE REVIEW

**Comment by alalek** [comment]
Proposed changes break MacOS builder videoio tests (with most fresh FFMPEG version installed):
```
--     FFMPEG:                      YES
--       avcodec:                   YES (ver 57.89.100)
--       avformat:                  YES (ver 57.71.100)
--       avutil:                    YES (ver 55.58.100)
--       swscale:                   YES (ver 4.6.100)
--       avresample:                YES (ver 3.5.0)
```

**Comment by jviney** [comment]
Tests passing again.

**Comment by alalek** [comment]
Are you able to run OpenCV's videoio tests with master branch of FFMPEG? (you need [opencv_extra repository](https://github.com/opencv/opencv_extra) and setup `OPENCV_TEST_DATA_PATH=<opencv_extra>/testdata` environment variable).

**Comment by jviney** [comment]
This patch works with the FFmpeg 3.3 and 3.4 branches, but not with current master (git `69218b419`). This assertion fails:

```
modules/videoio/test/test_ffmpeg.cpp:356: error: (-215) frameCount == WriteVideo_Invoker::FrameCount in function operator()
```

**Comment by alalek** [comment]
Thanks! I have the same observations about current master FFmpeg branch.

File: modules/videoio/src/cap_ffmpeg_impl.hpp [hunk 1]
```diff
@@ -1598,7 +1602,12 @@ static int icv_av_write_frame_FFMPEG( AVFormatContext * oc, AVStream * video_st,
 #endif
     int ret = OPENCV_NO_FRAMES_WRITTEN_CODE;
 
+#if LIBAVFORMAT_BUILD >= CALC_FFMPEG_VERSION(57, 0, 0)
+    if (video_st->codec->codec_type == AVMEDIA_TYPE_VIDEO &&
+        video_st->codec->codec_id == AV_CODEC_ID_RAWVIDEO) {
+#else
     if (oc->oformat->flags & AVFMT_RAWPICTURE) {
+#endif
```
Comment (alalek) [inline modules/videoio/src/cap_ffmpeg_impl.hpp]: `AVFMT_RAWPICTURE` have been deprecated/eliminated completely (there is no replacement for this), so code block under this "if" is not used at all anymore.

Please disable these code block completely:
```
#if LIBAVFORMAT_BUILD < CALC_FFMPEG_VERSION(57, 0, 0)
    if (oc->oformat->flags & AVFMT_RAWPICTURE) {
        /* raw video case. The API will change slightly in the near
           futur for that */
        AVPacket pkt;
        av_init_packet(&pkt);

#ifndef PKT_FLAG_KEY
#define PKT_FLAG_KEY AV_PKT_FLAG_KEY
#endif

        pkt.flags |= PKT_FLAG_KEY;
        pkt.stream_index= video_st->index;
        pkt.data= (uint8_t *)picture;
        pkt.size= sizeof(AVPicture);

        ret = av_write_frame(oc, &pkt);
    } else
#endif
    {
        /* encode the image */
        AVPacket pkt;
        av_init_packet(&pkt);
        ...
    }
```

Please apply similar changes for other `AVFMT_RAWPICTURE` entries (condition is negative).
```
#if LIBAVFORMAT_BUILD < CALC_FFMPEG_VERSION(57, 0, 0)
    if (!(oc->oformat->flags & AVFMT_RAWPICTURE))
#endif
    {
        ... code ...
    }
```

**Pull request merged**